viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -647,10 +647,31 @@ void SLAMServiceImpl::SaveMapWithTimestamp() {
     }
 }
 
+void SLAMServiceImpl::AddSensorReading(
+    cartographer::sensor::TimedPointCloudData measurement) {
+    cartographer::transform::Rigid3d tmp_global_pose;
+    bool update_latest_global_pose = false;
+
+    {
+        std::lock_guard<std::mutex> lk(map_builder_mutex);
+        if (measurement.ranges.size() > 0) {
```

Comment:
Why do we need to check that the measurement ranges's size is greater than 0?

Reply:
I am looking into this but having a tough time figuring out why - it is leftover from the old code

Reply:
I suspect this is here because, technically, a PCD could contain 0 points, which I suspect cartographer would complain about. This ensures that we only attempt to add measurements that contain data.